File: h/从上往下打印二叉树.java
/*
从上往下打印出二叉树的每个节点，同层节点从左至右打印。
*/
import java.util.ArrayList;
/**
public class TreeNode {
    int val = 0;
    TreeNode left = null;
    TreeNode right = null;

    public TreeNode(int val) {
        this.val = val;

    }

}
*/
import java.util.*;
public class Solution {
    public ArrayList<Integer> PrintFromTopToBottom(TreeNode root) {

        if (root == null) { 
            return new ArrayList<Integer>();
        }
        ArrayList<Integer> arr = new ArrayList<>();

        LinkedList<TreeNode> list = new LinkedList<>();
        list.add(root);
        while (!list.isEmpty()) {
            TreeNode temp = list.removeFirst();
            if (temp.left != null) {
                list.add(temp.left);
            }
            if (temp.right != null) {
                list.add(temp.right);
            }
            arr.add(temp.val);
        }
        return arr;
    }
}

File: z/从上往下打印二叉树.cc
/*
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    TreeNode(int x) :
            val(x), left(NULL), right(NULL) {
    }
};*/
class Solution {
 public:
  vector<int> PrintFromTopToBottom(TreeNode* root) {
    vector<int> res;
    if (root == NULL) {
      return res;
    }
    queue<TreeNode*> q;
    q.push(root);
    TreeNode* iter;
    while (!q.empty()) {
      int size = q.size();
      for (int i = 0; i < size; ++i) {
        iter = q.front();
        res.push_back(iter->val);
        q.pop();
        if (iter->left != NULL) {
          q.push(iter->left);
        }
        if (iter->right != NULL) {
          q.push(iter->right);
        }
      }
    }
    return res;
  }
};
